redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -25,133 +25,144 @@ import app.cash.redwood.Modifier
-    return get(index) ?: placeholder[index % placeholder.size]
-  }
+  override val value: UIView
+    get() = tableView
+
+  protected var onViewportChanged: ((firstVisibleItemIndex: Int, lastVisibleItemIndex: Int) -> Unit)? = null
+
+  private val processor = object : LazyListUpdateProcessor<LazyListContainerCell, UIView>() {
+    override fun createCell(
+      cell: Cell<LazyListContainerCell, UIView>,
+      widget: Widget<UIView>,
+      index: Int,
+    ): LazyListContainerCell {
+      val result = tableView.dequeueReusableCellWithIdentifier(
+        identifier = reuseIdentifier,
+        forIndexPath = NSIndexPath.indexPathForItem(index.convert(), 0.convert()),
+      ) as LazyListContainerCell
+      require(result.cell == null)
+      result.cell = cell
+      result.setWidget(widget.value)
+      return result
+    }
 
-  private val collectionViewDataSource: UICollectionViewDataSourceProtocol =
-    object : NSObject(), UICollectionViewDataSourceProtocol {
+    override fun setWidget(cell: LazyListContainerCell, widget: Widget<UIView>) {
+      cell.setWidget(widget.value)
+    }
 
-      override fun collectionView(
-        collectionView: UICollectionView,
-        numberOfItemsInSection: NSInteger,
-      ): NSInteger {
-        return items.size.toLong()
-      }
+    override fun insertRows(index: Int, count: Int) {
+      // TODO(jwilson): pass a range somehow when 'count' is large?
+      tableView.insertRowsAtIndexPaths(
+        (index until index + count).map { NSIndexPath.indexPathForItem(it.convert(), 0) },
+        UITableViewRowAnimationNone,
+      )
+    }
 
-      override fun collectionView(
-        collectionView: UICollectionView,
-        cellForItemAtIndexPath: NSIndexPath,
-      ): UICollectionViewCell {
-        val cell = collectionView.dequeueReusableCellWithReuseIdentifier(
-          identifier = reuseIdentifier,
-          forIndexPath = cellForItemAtIndexPath,
-        ) as LazyListContainerCell
+    override fun deleteRows(index: Int, count: Int) {
+      // TODO(jwilson): pass a range somehow when 'count' is large?
+      tableView.deleteRowsAtIndexPaths(
+        (index until index + count).map { NSIndexPath.indexPathForItem(it.convert(), 0) },
+        UITableViewRowAnimationNone,
+      )
+    }
+  }
 
-        cell.collectionViewFrame = collectionView.frame
-        cell.collectionViewFlowLayout = collectionViewFlowLayout
+  override val placeholder: Widget.Children<UIView> = processor.placeholder
 
-        val widget = items.itemForGlobalIndex(cellForItemAtIndexPath.item.toInt())
-        cell.set(widget.value)
+  override val items: Widget.Children<UIView> = processor.items
 
-        return cell
-      }
-    }
+  private val dataSource = object : NSObject(), UITableViewDataSourceProtocol {
+    override fun tableView(
+      tableView: UITableView,
+      numberOfRowsInSection: NSInteger,
+    ) = processor.size.toLong()
```

Comment:
nit: You might want to add an assertion here that `section == 0`

Reply:
Done.